src/Compilers/Core/VBCSCompiler/ServerDispatcher.cs

```diff
@@ -169,10 +168,6 @@ public void ListenAndDispatchConnections(string pipeName, TimeSpan? keepAlive, b
             Task<NamedPipeServerStream> listenTask = null;
             CancellationTokenSource listenCancellationTokenSource = null;
 
-            // If we aren't being asked to watch analyzer files then simple create a Task which never 
-            // completes.  This is the behavior of AnalyzerWatcher when files don't change on disk.
-            Task analyzerTask = watchAnalyzerFiles ? CompilerServerFileWatcher.CreateWatchFilesTask() : new TaskCompletionSource<bool>().Task;
```

Comment:
@mavasani The file watching code was attempting to solve a limited version of the same problem.

Say a compilation causes vbcscompiler.exe to load and run analyzer assembly A.dll. Before vbcscompiler.exe times out and shuts itself down, A.dll is updated on disk, and another compilation comes in that makes use of it. The new compilation will use the older-but-already-loaded A.dll, which is clearly not what was intended.

The file watcher tried to deal with this by noticing that A.dll had changed, and killing vbcscompiler.exe before future builds would accidentally make use of the old one.

Now instead of kill vbcscompiler.exe we'll reject the new submission because updating A.dll will presumable cause the MVID to change.